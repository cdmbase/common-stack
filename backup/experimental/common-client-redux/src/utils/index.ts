export * from './action-helpers';
export * from './context-middleware';
