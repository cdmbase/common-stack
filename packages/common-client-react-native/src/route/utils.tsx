import * as React from 'react';
import { IRoute as IRouteProps, IRouteComponentProps } from '@common-stack/client-react';
import { IScreen,  } from '../interfaces';
import { nanoid } from 'nanoid/non-secure';
import { Redirect } from 'react-router-config';

export function flattenRoutes(routes?: IRouteProps[], parent?: IScreen): IScreen[] {
    if (!Array.isArray(routes)) return [];
    const screens: IScreen[] = [];
    for (let idx = 0; idx < routes.length; idx++) {
        const route = routes[idx];
        const { key: routeKey, path, exact, component, strict, redirect, routes: children, ...options } = route;
        const screenKey = routeKey || nanoid();
        const screen: IScreen = {
            key: screenKey,
            name: path || '/',
            component: function ScreenComponent(props) {
                if (redirect) {
                    return <Redirect from={path} to={redirect} exact={exact} strict={strict} />;
                }
                const children = component ? React.createElement(component, props) : null;
                return parent && parent.component ? React.createElement(parent.component, props, children) : children;
            },
            options: {
                ...options,
                routeMatchOpts: route,
            },
        };
        if (Array.isArray(children) && children.length > 0) {
            screens.push(...flattenRoutes(children, screen));
        } else {
            screens.push(screen);
        }
    }
    return screens;
}

