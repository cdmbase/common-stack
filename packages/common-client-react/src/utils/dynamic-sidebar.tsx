import * as React from 'react';
import { Query, withApollo } from 'react-apollo';
import gql from 'graphql-tag';
import Course from '../connector/course';

export function renderQuery(loading, error, data) {
  if (loading) return <p>Loading...</p>;
  if (error) return <p>Error :(</p>;
  return data.allPageHistories.map(allHistory => (
    <Course key={allHistory.id} history={allHistory} />
  ));
}

function AddHistory(props) {
  const { mutate } = props.client;
  const { push, pageNumber } = props;
  mutate({ mutation: gql`
    mutation {
      createPageHistory(number: ${pageNumber}) {
        number
      }
    },
    `, refetchQueries: [{ query: gql`
          {
            allPageHistories(orderBy: id_DESC, first: 5) {
              id
              number
            }
          }
        ` }] }).catch(() => {});
  push(`/history/${pageNumber}`);
  return null;
};

export class History extends React.Component {
  render() {
    return (
      <Query
        query={gql`
          {
            allPageHistories(orderBy: id_DESC, first: 5) {
              id
              number
            }
          }
        `}
      >
        {({ loading, error, data }) => renderQuery(loading, error, data)}
      </Query>
    );
  }
}

export default withApollo(AddHistory as any);

