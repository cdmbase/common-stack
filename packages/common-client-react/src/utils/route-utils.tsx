import * as React from 'react';
import { IRouteData, IMappedData, IMenuData, IMenuItem, IMenuPosition } from '../interfaces';
import { RouteProps, Switch } from 'react-router';
import { Route } from 'react-router-dom';
const sortKeys = require('sort-keys');


/* eslint no-useless-escape:0 */
const reg = /(((^https?:(?:\/\/)?)(?:[-;:&=\+\$,\w]+@)?[A-Za-z0-9.-]+|(?:www.|[-;:&=\+\$,\w]+@)[A-Za-z0-9.-]+)((?:\/[\+~%\/.\w-_]*)?\??(?:[-\+=&;%@.\w_]*)#?(?:[\w]*))?)$/g;

export function isUrl(path) {
  return reg.test(path);
}
/**
 * Generates Routes based on the key value, where key has the path of the route and value
 * has rest of the values for building a `<Router ../>` component.
 *
 * @param path: RegExp
 * @param routerData
 */
export function getRoutes2(path: RegExp, routerData: IRouteData) {
  const routes = Object.keys(routerData).filter(routePath => {
    return routePath.match(path);
  });
  const mappedRoutes: Array<IMappedData> = routes.map(paths => {
    return {
      route: paths,
      ...routerData[paths],
    };
  });
  const root: RouteProps & { routes?: any } = {
  };
  mappedRoutes.forEach(eachRoute => {
    const children = eachRoute.route.split('/');
    children.shift();

    // if the route is `/` then add this correction.
    if (eachRoute.route === '/') {
      children.shift();
      children.push('/');
    }
    const depth = children.length;
    let lastNode = root;
    for (let i = 0; i <= depth; i++) {
      const lastIndex = (lastNode.routes || []).findIndex(item => {
        if (eachRoute.route.startsWith(item.path)) {
          return eachRoute.route[item.path.length] === '/';
        }
      });
      if (lastIndex === -1) {
        break;
      }
      lastNode = lastNode.routes[lastIndex];
      // do not overwrite the route's exact value
      lastNode.exact = routerData[lastNode.path as string].hasOwnProperty('exact') ? routerData[lastNode.path as string].exact : false;
    }
    if (!lastNode.routes) {
      lastNode.routes = [];
    }
    const modfiedRoute = formatSlash(eachRoute.route);
    lastNode.routes.push({
      ...eachRoute,
      // route: modfiedRoute,
      path: modfiedRoute,
      exact: routerData[eachRoute.route].hasOwnProperty('exact') ? routerData[eachRoute.route].exact : true,
      component: eachRoute.component,
    });
  });

  return root.routes;
}

const startWithMoreThanOneSlash = /^(\/)\1+/; // for exame `//abc, ///abc, ...`
const formatSlash = (route) => {
  // replaced `//` with `/` in the routes.
  return {
    path: route.replace(startWithMoreThanOneSlash, '/'),
    _pathPrefix: (route.match(/^\/(\/{1,})/) || ['', ''])[1],
  } 
}

export function getRoutes(path: string, routeData: IRouteData) {
  if (!path.startsWith('/')) {
    throw new Error('Invalid path!');
  }
  let searchPath = path;
  if (path[path.length - 1] !== '/') {
    searchPath = path + '/';
  }
  const routes = Object.keys(routeData).filter(menuPath => {
    return menuPath.indexOf(searchPath) === 0 || menuPath === path;
  });

  const mappedMenuPaths: Array<IMappedData> = routes.map(mPath => {
    return {
      route: mPath,
      position: IMenuPosition.MIDDLE,
      ...routeData[mPath],
    };
  });
  const root: RouteProps & { routes?: any } = {
    // just to satisfy types added following
    // TOOD need to correct types so we don't have to enter them.
    name: 'root',
    position: IMenuPosition.LOGO,
  } as any;
  mappedMenuPaths.forEach(routeItem => {
    const children = routeItem.route.split('/');
    children.shift();

    // if the route is `/` then add this correction.
    if (routeItem.path === '/') {
      children.shift();
      children.push('/');
    }
    const depth = children.length;
    let lastNode = root;
    for (let i = 0; i < depth; i++) {
      const lastIndex = (lastNode.routes || []).findIndex(item => {
        const routePath = `${item._pathPrefix}${item.path}`;

        if (routeItem.route.startsWith(routePath)) {
          return routeItem.route[routePath.length] === '/';
        }
      });
      if (lastIndex === -1) {
        break;
      }
      lastNode = lastNode.routes[lastIndex];
      const lastNodePath = `${(lastNode as any)._pathPrefix}${lastNode.path}`;
      lastNode.exact = routeData[lastNodePath].hasOwnProperty('exact') ? routeData[lastNodePath].exact : false;
    }
    if (!lastNode.routes) {
      lastNode.routes = [];
    }
    const { route: ignore, ...rest } = routeItem;
    const pathParams = formatSlash(routeItem.route);
    lastNode.routes.push({
      ...rest,
      // path: formatSlash(routeItem.route),
      // path: routeItem.route,
      ...pathParams,
      exact: routeData[routeItem.route].hasOwnProperty('exact') ? routeData[routeItem.route].exact : true,
      component: routeItem.component,
    });
  });

  return root.routes;
}
export function getMenus(path: string, menuData: IMenuData) {
  if (!path.startsWith('/')) {
    throw new Error('Invalid path!');
  }
  let searchPath = path;
  if (path[path.length - 1] !== '/') {
    searchPath = path + '/';
  }
  const routes = Object.keys(menuData).filter(menuPath => {
    return menuPath.indexOf(searchPath) === 0 || menuPath === path;
  });

  const mappedMenuPaths: Array<IMenuItem & { route?: string }> = routes.map(mPath => {
    return {
      route: mPath,
      position: IMenuPosition.MIDDLE,
      ...menuData[mPath],
    };
  });
  const root: IMenuItem & { route?: string } = {
    // just to satisfy types added following
    // TOOD need to correct types so we don't have to enter them.
    name: 'root',
    position: IMenuPosition.LOGO,
  };
  mappedMenuPaths.forEach(menutItem => {
    const children = menutItem.route.split('/');
    children.shift();
    const depth = children.length;
    let lastNode = root;
    for (let i = 0; i < depth; i++) {
      const lastIndex = (lastNode.children || []).findIndex(item => {
        if (menutItem.route.startsWith(item.path)) {
          return menutItem.route[item.path.length] === '/';
        }
      });
      if (lastIndex === -1) {
        break;
      }
      lastNode = lastNode.children[lastIndex];
      // do not overwrite the route's exact value
      lastNode.exact = menuData[lastNode.path].hasOwnProperty('exact') ? menuData[lastNode.path].exact : false;
    }
    if (!lastNode.children) {
      lastNode.children = [];
    }
    const { route: ignore, ...rest } = menutItem;
    lastNode.children.push({
      path: menutItem.route,
      ...rest,
      exact: menuData[menutItem.route].hasOwnProperty('exact') ? menuData[menutItem.route].exact : true,
    });
  });

  return root.children;
}


export const renderRoutes = (routes, solidRoutes, extraProps = {}, switchProps = {}) =>
  routes ? (
    <>
      {[
        ...solidRoutes, ...routes.map((route, i) => (
          <Route
            key={route.key || i}
            path={route.path}
            exact={route.exact}
            strict={route.strict}
            render={props =>
              route.render ? (
                route.render({ ...props, ...extraProps, route: route })
              ) : (
                  <route.component {...props} {...extraProps} route={route} />
                )
            }
          />
        )),
      ]}
    </>
  ) : null;


export const getSortedRoutes = (path: string, routeData: IRouteData) => {
  const sortedRoutes = sortKeys(routeData, { compare });
  return getRoutes(path, sortedRoutes);
};
const compare = ((a, b) => {
  const aStr = String(a).toLowerCase();
  const bStr = String(b).toLowerCase();

  // Alphanumeric elements always come before non-alphanumeric elements
  const aIsAlphanumeric = notStartWithColon(aStr);
  const bIsAlphanumeric = notStartWithColon(bStr);
  if (aIsAlphanumeric + bIsAlphanumeric !== -2) {
    if(aIsAlphanumeric * bIsAlphanumeric > 0){
      if (aIsAlphanumeric === bIsAlphanumeric){
        return aStr.localeCompare(bStr);
      }
      return aIsAlphanumeric - bIsAlphanumeric;
    }
    if (aIsAlphanumeric > 0) {
      return -1;
    } else if (bIsAlphanumeric > 0) {
      return 1;
    }
  }

  // Numerical elements always come before alphabetic elements
  const aNum = Number(a);
  const bNum = Number(b);
  // If both are numerical, sort in the usual fashion (smaller goes first)
  if (aNum && bNum) {
    return aNum - bNum;
    // If a is numerical but b isn't, put a first.
  } else if (aNum) {
    return -1;
    // If b is numerical but a isn't, put b first.
  } else if (bNum) {
    return 1;
  }

  // In all other cases, default to usual sort order.
  return aStr.localeCompare(bStr);
});


function notStartWithColon(str) {
  const match = (str.match(/^[\/, *, :]{1,}/) || [-1])[0];
  if (match === -1) {
    return -1;
  }
  var i = match.length;
  let count = 0;
  while (i--) {
    const char = match[i];
    if (char === '/') {
      count = count + 3;
    } else if (char === ':') {
      count = count + 1;
    } else if (char === '*') {
      count = count + 2;
    }
  }
  return count;
}