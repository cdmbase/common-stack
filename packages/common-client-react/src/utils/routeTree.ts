import { IRouteData } from '../interfaces';

export default class RouteTree implements IRouteData {
	public root: IRouteData;
	public addStatus: boolean;

	constructor(node: IRouteData) {
		this.root = node;
	}

	public traverse(callback: () => any) {
		function walk(node: IRouteData) {
			callback(node);
			node.routes.forEach(walk);
		}

		walk(this.root);
	}

	public addNode(path: String, addlData: Object, parentPath: Object) {
		let newNode = {
			path,
			...addlData,
			routes: [],
		};

		this.addStatus = false;

		if (this.root === null) {
			this.root = newNode;
			return;
		}

		this.traverse(node => {
			if (node.path === parentPath) {
				node.routes.push(newNode);
				this.addStatus = true;
			}
		});
	}
}
