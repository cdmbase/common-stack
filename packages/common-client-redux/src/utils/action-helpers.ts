function isFunction(obj) {
    return !!(obj && obj.constructor && obj.call && obj.apply)
}

export const addToAction = ($action, extend: any) =>
    isFunction($action) 
        ? Object.assign({ $action }, extend) 
        : Object.assign(extend, $action);

export const sideAction = () => (action: any) => addToAction(action, { $side: true });
export const withContext = (...$dependencies: string[]) => (action: any) => addToAction(action, { $dependencies });

export const createSideAction = (action) => (...args) => sideAction()(action(...args));
export const createActionWithContext = (action, context) => (...args) => withContext(...context)(action(...args));
