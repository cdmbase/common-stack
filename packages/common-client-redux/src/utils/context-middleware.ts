export interface IContextResolver {
    [propName: string]: (action: any) => Promise<any>;
}

export const contextMiddleware = (contextResolvers: any) => {
    return ({ dispatch, getState }) => next => action => {
        let act = action;
        let ctxPromise = Promise.resolve({});

        if (action['$dependencies'] && Array.isArray(action['$dependencies'])) {            
            let deps = action.$dependencies.map(dep => Promise.resolve(contextResolvers[dep].call(undefined, action)).then(value => ({ dep, value })));
            ctxPromise = Promise.all(deps) 
                .then((dependencies: any[]) => 
                    dependencies.reduce((acc, { dep, value }) => Object.assign(acc, { [dep]: value }), {}),
                );
        }

        ctxPromise.then(context => {
            act = action.$action || act;
            if (typeof act === 'function') {
                return act(dispatch, getState, context);
            }
    
            return next({ ...action, context });
        });
    };
};
