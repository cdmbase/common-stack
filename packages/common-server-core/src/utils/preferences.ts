import { IPreferncesTransformed, IPreferences } from '../interfaces';

export const getCurrentPreferences = (preferencesDefaultArr, overwritePreferences) => {
  const preferencesArr = { ...preferencesDefaultArr };

  Object.keys(preferencesDefaultArr).forEach(key => {
    if (preferencesDefaultArr[key].overridable) {
      const overwritesSettings = Object.keys(overwritePreferences).filter(el => el.indexOf(key) !== -1);
      let newValues = {};
      overwritesSettings.forEach(item => {
        const overwriteKey = item.split('.').pop();
        newValues[overwriteKey] = overwritePreferences[item];
      });
      preferencesArr[key] = { ...preferencesDefaultArr[key], ...newValues };
    }
  });
  return preferencesArr;
};

export const transformPrefsToArray = (preferences: IPreferences) => {
  const resultArr: IPreferncesTransformed[] = [];
  Object.keys(preferences).forEach(key => {
    const type = key.split('.')[0];
    const findType = resultArr.find(el => el.type === type);
    if (!findType) {
      resultArr.push({
        type,
        data: [{
          name: key,
          ...preferences[key],
        }],
      });
    } else {
      findType['data'].push({
        name: key,
        ...preferences[key],
      });
    }
  });
  return resultArr;
};
