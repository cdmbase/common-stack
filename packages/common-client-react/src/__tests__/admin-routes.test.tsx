import * as React from 'react';
import { getSortedRoutes, renderRoutes, getMenus } from '../utils';



function MyComponent(props) {
    return <div>MyComponent {JSON.stringify(props)}</div>;
}

const routerConfig1 = {
    ['/']: {
        exact: false,
        component: MyComponent,
        name: 'rootSlash',
    },
    ['//']: {
        exact: true,
        component: MyComponent,
        name: 'doubleShlash'
    },
    ['//a']: {
        component: MyComponent,
        name: 'doubleSlashA'
    },
    ['//a/1']: {
        component: MyComponent,
        name: 'doubleShlashA1'

    },
    ['//a/:b']: {
        component: MyComponent,
        name: 'doubleShlashAB'

    },
    ['//a/:b/1']: {
        component: MyComponent,
        name: 'doubleShlashAB1'

    },
    ['//a/:b/c']: {
        exact: true,
        component: MyComponent,
        name: 'doubleShlashABC'

    },
    ['//a/:b/c/1']: {
        component: MyComponent,
        name: 'doubleShlashABC1'

    },
    ['//a/:c/1']: {
        component: MyComponent,
        name: 'doubleShlashAC1'

    },
    ['/a/2']: {
        component: MyComponent,
        name: 'doubleShlashA2'

    },
    ['/a/2/1']: {
        component: MyComponent,
        name: 'doubleShlashA21'

    },
    ['/ab/2/1']: {
        component: MyComponent,
        name: 'doubleShlashAB21'

    },
    ['/b/1']: {
        component: MyComponent,
        name: 'doubleShlash21'

    },
    ['/b/login/register']: {
        component: MyComponent,
        name: 'doubleShlashBLogin'

    },
};
const orgRoutes = {
    ["/login"]: { exact: true, component: MyComponent },
    ["/callback"]: { exact: true, component: MyComponent },
    ["/:orgName/usermenu/logout"]: { exact: true, component: MyComponent },
    ["/:orgName/usermenu/profile"]: { exact: true, component: MyComponent },
    ["/authenticate-popup"]: { exact: true, component: MyComponent },
    ["/reset-password"]: { exact: true, component: MyComponent },
    ["/:orgName/usermenu/account/"]: { exact: false, component: MyComponent },
    ["/:orgName/usermenu/billing"]: { exact: false, component: MyComponent },
    ["/"]: { exact: true, component: MyComponent },
    ["/:orgName/usermenu/account"]: { exact: false, component: MyComponent },
    ["/:orgName/usermenu/account-settings"]: { exact: true, component: MyComponent },
    ["/:orgName/teams/view/:teamName"]: { exact: false, component: MyComponent },
    ["/:orgName/teams"]: { exact: true, component: MyComponent },
    ["/:orgName/invitation"]: { exact: true, component: MyComponent },
    ["/:orgName/usermenu/organizations"]: { exact: true, component: MyComponent },
    ["/:orgName/usermenu/switch-organization"]: { exact: true, component: MyComponent },
    ["/:orgName/pref"]: { exact: false, component: MyComponent },
    ["/:orgName/members"]: { exact: false, component: MyComponent },
    ["/:orgName/dashboard"]: { exact: true, component: MyComponent },
    ["/:orgName/registries"]: { exact: true, component: MyComponent },
    ["/:orgName/create-workspace"]: { exact: true, component: MyComponent },
    ["/:orgName/workspace/:id/details"]: { exact: true, component: MyComponent },
    ["/:orgName/workspace/:id/editor"]: { exact: true, component: MyComponent },
    ["/:orgName/workspace/:id"]: { exact: true, component: undefined },
    ["/:orgName/repository"]: { exact: true, component: MyComponent },
    ["/:orgName/artifact"]: { exact: false, component: MyComponent },
    ["/:orgName"]: { exact: false, component: MyComponent },
};

const layoutRoutes = {
    ['/']: {
        exact: false,
        component: MyComponent,
        name: 'rootSlash',
    },
    ["/:test"]: { exact: true, component: MyComponent },
    ["//home"]: { exact: true, component: MyComponent },
    ["/callback"]: { exact: true, component: MyComponent },
    ["//:orgName/usermenu/logout"]: { exact: true, component: MyComponent },
    ["//:orgName/usermenu/profile"]: { exact: true, component: MyComponent },
    ["/authenticate-popup"]: { exact: true, component: MyComponent },
    ["/reset-password"]: { exact: true, component: MyComponent },
    ["//:orgName/usermenu/account/"]: { exact: false, component: MyComponent },
    ["//:orgName/usermenu/billing"]: { exact: false, component: MyComponent },
    ["//"]: { exact: false, component: MyComponent },
    ["//:orgName/usermenu/account"]: { exact: false, component: MyComponent },
    ["//:orgName/usermenu/account-settings"]: { exact: true, component: MyComponent },
    ["//:orgName/teams/view/:teamName"]: { exact: false, component: MyComponent },
    ["//:orgName/teams"]: { exact: true, component: MyComponent },
    ["//:orgName/invitation"]: { exact: true, component: MyComponent },
    ["//:orgName/usermenu/organizations"]: { exact: true, component: MyComponent },
    ["//:orgName/usermenu/switch-organization"]: { exact: true, component: MyComponent },
    ["//:orgName/pref"]: { exact: false, component: MyComponent },
    ["//:orgName/members"]: { exact: false, component: MyComponent },
    ["//:orgName/dashboard"]: { exact: true, component: MyComponent },
    ["//:orgName/registries"]: { exact: true, component: MyComponent },
    ["//:orgName/create-workspace"]: { exact: true, component: MyComponent },
    ["//:orgName/workspace/:id/details"]: { exact: true, component: MyComponent },
    ["//:orgName/workspace/:id/editor"]: { exact: true, component: MyComponent },
    ["//:orgName/workspace/:id"]: { exact: true, component: undefined },
    // ["//:orgName/repository"]: { exact: true, component: MyComponent },
    // ["//:orgName/artifact"]: { exact: false, component: MyComponent },
    // ["/:orgName"]: { exact: false, component: MyComponent },
};

describe('getRoutes utility with basic routes', () => {


    xtest('with /a', async () => {
        const result = {
            // component: any,
            routes: [{
                path: '/a', routes:
                    [{ path: '/a/1', exact: true },
                    { path: '/a/:b/1', exact: true },
                    { path: '/a/2', routes: [{ 'exact': true, 'path': '/a/2/1' }] },
                    { path: '/ab/2/1', exact: true }],
            }],
        };
        const routes1 = getSortedRoutes('/', routerConfig1);
        // const routes = getRoutes(/^\/.*/, sortKeys(routerConfig, { compare }));
        const menu1 = getMenus('/', routes1);
        expect(routes1).toMatchSnapshot();
        // expect(menu1).toMatchSnapshot();
        // expect(routes).toMatchSnapshot();

    });

    xtest('more url testing', () => {
        const routes = getSortedRoutes('/', orgRoutes);
        expect(routes).toMatchSnapshot();

    })

    test('layout routes', () => {
        const routes = getSortedRoutes('/', layoutRoutes);
        expect(routes).toMatchSnapshot();
    });

    xtest('render /a', async () => {
        const routes1 = getSortedRoutes('/', routerConfig1);
        const renderedRoutes = renderRoutes(routes1, []);
        expect(renderedRoutes).toMatchSnapshot();
    })

});
