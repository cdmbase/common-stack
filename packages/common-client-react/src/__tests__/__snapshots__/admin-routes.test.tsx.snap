// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getRoutes utility with basic routes layout routes 1`] = `
Array [
  Object {
    "_pathPrefix": "",
    "component": [Function],
    "exact": false,
    "name": "rootSlash",
    "path": "/",
    "position": "MIDDLE",
    "routes": Array [
      Object {
        "_pathPrefix": "/",
        "component": [Function],
        "exact": false,
        "path": "/",
        "position": "MIDDLE",
      },
      Object {
        "_pathPrefix": "/",
        "component": [Function],
        "exact": true,
        "path": "/home",
        "position": "MIDDLE",
      },
      Object {
        "_pathPrefix": "/",
        "component": [Function],
        "exact": true,
        "path": "/:orgName/create-workspace",
        "position": "MIDDLE",
      },
      Object {
        "_pathPrefix": "/",
        "component": [Function],
        "exact": true,
        "path": "/:orgName/dashboard",
        "position": "MIDDLE",
      },
      Object {
        "_pathPrefix": "/",
        "component": [Function],
        "exact": true,
        "path": "/:orgName/invitation",
        "position": "MIDDLE",
      },
      Object {
        "_pathPrefix": "/",
        "component": [Function],
        "exact": false,
        "path": "/:orgName/members",
        "position": "MIDDLE",
      },
      Object {
        "_pathPrefix": "/",
        "component": [Function],
        "exact": false,
        "path": "/:orgName/pref",
        "position": "MIDDLE",
      },
      Object {
        "_pathPrefix": "/",
        "component": [Function],
        "exact": true,
        "path": "/:orgName/registries",
        "position": "MIDDLE",
      },
      Object {
        "_pathPrefix": "/",
        "component": [Function],
        "exact": true,
        "path": "/:orgName/teams",
        "position": "MIDDLE",
        "routes": Array [
          Object {
            "_pathPrefix": "/",
            "component": [Function],
            "exact": false,
            "path": "/:orgName/teams/view/:teamName",
            "position": "MIDDLE",
          },
        ],
      },
      Object {
        "_pathPrefix": "/",
        "component": [Function],
        "exact": false,
        "path": "/:orgName/usermenu/account",
        "position": "MIDDLE",
        "routes": Array [
          Object {
            "_pathPrefix": "/",
            "component": [Function],
            "exact": false,
            "path": "/:orgName/usermenu/account/",
            "position": "MIDDLE",
          },
        ],
      },
      Object {
        "_pathPrefix": "/",
        "component": [Function],
        "exact": true,
        "path": "/:orgName/usermenu/account-settings",
        "position": "MIDDLE",
      },
      Object {
        "_pathPrefix": "/",
        "component": [Function],
        "exact": false,
        "path": "/:orgName/usermenu/billing",
        "position": "MIDDLE",
      },
      Object {
        "_pathPrefix": "/",
        "component": [Function],
        "exact": true,
        "path": "/:orgName/usermenu/logout",
        "position": "MIDDLE",
      },
      Object {
        "_pathPrefix": "/",
        "component": [Function],
        "exact": true,
        "path": "/:orgName/usermenu/organizations",
        "position": "MIDDLE",
      },
      Object {
        "_pathPrefix": "/",
        "component": [Function],
        "exact": true,
        "path": "/:orgName/usermenu/profile",
        "position": "MIDDLE",
      },
      Object {
        "_pathPrefix": "/",
        "component": [Function],
        "exact": true,
        "path": "/:orgName/usermenu/switch-organization",
        "position": "MIDDLE",
      },
      Object {
        "_pathPrefix": "/",
        "component": undefined,
        "exact": true,
        "path": "/:orgName/workspace/:id",
        "position": "MIDDLE",
        "routes": Array [
          Object {
            "_pathPrefix": "/",
            "component": [Function],
            "exact": true,
            "path": "/:orgName/workspace/:id/details",
            "position": "MIDDLE",
          },
          Object {
            "_pathPrefix": "/",
            "component": [Function],
            "exact": true,
            "path": "/:orgName/workspace/:id/editor",
            "position": "MIDDLE",
          },
        ],
      },
    ],
  },
  Object {
    "_pathPrefix": "",
    "component": [Function],
    "exact": true,
    "path": "/authenticate-popup",
    "position": "MIDDLE",
  },
  Object {
    "_pathPrefix": "",
    "component": [Function],
    "exact": true,
    "path": "/callback",
    "position": "MIDDLE",
  },
  Object {
    "_pathPrefix": "",
    "component": [Function],
    "exact": true,
    "path": "/reset-password",
    "position": "MIDDLE",
  },
  Object {
    "_pathPrefix": "",
    "component": [Function],
    "exact": true,
    "path": "/:test",
    "position": "MIDDLE",
  },
]
`;
